mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/server/master/scheduler/ResourceClusterAwareScheduler.java

```diff
@@ -37,8 +38,13 @@ public class ResourceClusterAwareScheduler implements MantisScheduler {
     private final ActorRef schedulerActor;
 
     @Override
-    public void scheduleWorker(ScheduleRequest scheduleRequest) {
-        schedulerActor.tell(ScheduleRequestEvent.of(scheduleRequest), null);
+    public void scheduleWorkers(BatchScheduleRequest scheduleRequest) {
+        schedulerActor.tell(BatchScheduleRequestEvent.of(scheduleRequest), null);
+    }
+
+    @Override
+    public void unscheduleJob(String jobId) {
+        schedulerActor.tell(CancelBatchRequestEvent.of(jobId),null);
```

Comment:
CancelBatchRequestEvent seems to be only cancelling retrying on the scheduler side. I am wondering what happens if the rcActor is sending the assignment request while this is happening? Will the agent still get the cancel request?

Reply:
Not sure I understand the question. If you're asking what happens if the user decides to kill the job, there are 2 cases: 
1) job is in pending state because not enough resources are available. In this case we'll simply delete the timer so the scheduler won't retry. 
2) job is in accepted state and workers are already assigned. Those will still get cancelled as we're currently doing via `unscheduleAndTerminateWorker`.